friendshipper/src-tauri/src/repo/operations/gh/submit.rs

```diff
@@ -220,53 +219,38 @@ where
 
         // save a snapshot before submitting
         // make sure we have a temp dir for copying our files
-        let proj_dirs =
-            ProjectDirs::from("", "", APP_NAME).ok_or(anyhow!("Failed to get project dirs"))?;
-
-        let mut stash_path = proj_dirs.data_dir().to_path_buf();
-        stash_path.push(".stash");
-
-        // reset the stash path, delete and recreate
-        if stash_path.exists() {
-            std::fs::remove_dir_all(&stash_path)?;
-        }
-
-        std::fs::create_dir_all(&stash_path)?;
-
-        let current_branch = self.git_client.current_branch().await?;
-
-        // identify files that were deleted in the request
-        let (stashed_files, deleted_files): (Vec<String>, Vec<String>) =
-            self.files.iter().cloned().partition(|file| {
-                // build path from repo path
-                let path = self.git_client.repo_path.join(file);
-                // check if the file exists
-                path.exists()
-            });
-
-        // for each file, copy it to the stash
-        for file in &stashed_files {
-            let src = self.git_client.repo_path.join(file.clone());
-            let dest = stash_path.join(file);
-
-            std::fs::create_dir_all(dest.parent().unwrap())?;
-            std::fs::copy(src, dest)?;
-        }
+        let snapshot = self
+            .git_client
+            .save_snapshot_all("pre-submit", SaveSnapshotIndexOption::KeepIndex)
+            .await?;
 
         match self.execute_internal().await {
             Ok(_) => Ok(()),
             Err(e) => {
                 // attempt to reset to original branch and restore snapshot
                 // if this fails for any reason, we should simply log, then return the original error
-                let _ = self
-                    .recover(stash_path, &current_branch, stashed_files, deleted_files)
+                let modified_files = self.repo_status.read().modified_files.clone();
+                let untracked_files = self.repo_status.read().untracked_files.clone();
+                let all_files: Vec<File> = modified_files
+                    .0
+                    .into_iter()
+                    .chain(untracked_files.0.into_iter())
+                    .collect();
+
+                match self
+                    .git_client
+                    .restore_snapshot(&snapshot.commit, all_files)
                     .await
-                    .map_err(|err| {
-                        warn!(
-                            "Failed to recover from error during submit operation: {}",
-                            err
-                        );
-                    });
+                {
+                    Ok(_) => {
+                        // if we successfully restored the snapshot, we should be able to reset to the original branch
+                        let branch = self.repo_status.read().branch.clone();
+                        self.git_client.hard_reset(&branch).await?;
```

Comment:
won't this wipe out all the changes that just got restored? do we want to do this _before_ calling `restore_snapshot`?

Reply:
hahahaha yeah.....